go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -124,64 +129,31 @@ func (oc *DefaultNetworkController) newEgressFirewallRule(rawEgressFirewallRule
 // The following cleanups are needed from the previous versions:
 // - 	Cleanup the old implementation (using LRP) in local GW mode
 //  	For this it just deletes all LRP setup done for egress firewall
-//  	And also convert all old ACLs which specifed from-lport to specifying to-lport
 
 // -	Cleanup the old implementation (using ACLs on the join and node switches)
 //  	For this it deletes all the ACLs on the join and node switches, they will be created from scratch later.
 
 // NOTE: Utilize the fact that we know that all egress firewall related setup must have a priority: types.MinimumReservedEgressFirewallPriority <= priority <= types.EgressFirewallStartPriority
 // Upon failure, it may be invoked multiple times in order to avoid a pod restart.
 func (oc *DefaultNetworkController) syncEgressFirewall(egressFirewalls []interface{}) error {
-	// Lookup all ACLs used for egress Firewalls
-	aclPred := func(item *nbdb.ACL) bool {
-		return item.Priority >= types.MinimumReservedEgressFirewallPriority && item.Priority <= types.EgressFirewallStartPriority
-	}
-	egressFirewallACLs, err := libovsdbops.FindACLsWithPredicate(oc.nbClient, aclPred)
-	if err != nil {
-		return fmt.Errorf("unable to list egress firewall ACLs, cannot cleanup old stale data, err: %v", err)
-	}
-
-	// Update the direction of each egressFirewallACL if needed.
-	// Update the egressFirewallACL name if needed.
-	namespaceToACLs := map[string][]*nbdb.ACL{}
-	for i := range egressFirewallACLs {
-		egressFirewallACLs[i].Direction = nbdb.ACLDirectionToLport
-		if namespace, ok := egressFirewallACLs[i].ExternalIDs[egressFirewallACLExtIdKey]; ok && namespace != "" {
```

Comment:
are you removing all of this (line 135-162) because we no longer need to update the direction as it has already been done in older versions of OCP?

are lines 164-174 removed because you handle updating the logging in updateStaleEgressFirewallACLs?

Reply:
with this update the logic of a sync changes a bit: now we make sure the ExternalIDs are correct, then every initial Add function will just update existing acl to the latest version of what it would create. Meaning that we don't need to change all the fields manually, but just update ExternalIDs to make sure ACLs will be properly updated